sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/statefulOperators.scala

```diff
@@ -75,7 +76,9 @@ trait StatefulOperator extends SparkPlan {
 
   // Function used to record state schema for the first time and validate it against proposed
   // schema changes in the future. Runs as part of a planning rule on the driver.
-  def validateAndMaybeEvolveStateSchema(hadoopConf: Configuration): Unit
+  // Returns the schema file path for operators that write this to the metadata file,
+  // otherwise None
+  def validateAndMaybeEvolveStateSchema(hadoopConf: Configuration, batchId: Long): Option[String]
```

Comment:
for join - a single string might not be enough right ? we need a path for each store name ?